pymc/distributions/mixture.py

```diff
@@ -367,6 +373,25 @@ def marginal_mixture_logprob(op, values, rng, weights, *components, **kwargs):
     return mix_logp
 
 
+@_get_moment.register(MarginalMixtureRV)
+def get_moment_marginal_mixture(op, rv, rng, weights, *components):
+    if len(components) == 1:
+        ndim_supp = components[0].owner.op.ndim_supp
+
+        # assuming weights.shape[-1] is the mixing component
+        # trying to expand dimensions by ndim_supp inside out
+        expansion_axes = [-i for i in range(ndim_supp, 0, -1)]
+
+        moment_components = get_moment(components[0])
+
+        weights = at.expand_dims(weights, axis=expansion_axes)
```

Comment:
Might be more readable to do:
```suggestion
        weights = at.shape_padright(weights, ndim_supp)        
        moment_components = get_moment(components[0])
```